### ISSUE TITLE
Buildbot website not working under Safari due to WebSocket implementation not handling HTTP 401

### ISSUE THREAD

**Issue by krf** [buildbot/buildbot#3570]
The Buildbot web site does not function properly under Safari. The issue the web inspector reports is:

```
`WebSocket connection to 'wss://ci.example.com/ws' failed: Unexpected response code: 401
```

The underlying problem seems to be a bug in Safari/WebKit itself:
https://bugs.webkit.org/show_bug.cgi?id=80362

Screenshot:
![buildbot-safari-bug](https://user-images.githubusercontent.com/30826/29968259-17719dc8-8f1b-11e7-9836-a5580f2f7311.png)

**Comment by tardyp** [comment]
What is you authentication setup? I doubt 80362 can apply to Buildbot..

**Comment by lancelodliu** [comment]
This problem happened to me too.

**Comment by tardyp** [comment]
@krf @lancelodliu  nine.buildbot.net works for me under safari. You need to be more detailled on your report so that I can help.

Buildbot will never return 401 on /ws so I would guess this is a problem with the reverse proxy.

As a workaround you may want to allow all on /ws

**Comment by seankelly** [comment]
Any proxy?

**Comment by krf** [comment]
Yes, we're using an Apache reverse proxy which supports HTTP-based authentication. Via Buildbot's `util.RemoteUserAuth`.

**Comment by rizumu** [comment]
I worked around this error by editing my nginx config to support websockets. See discussion here: https://github.com/websockets/ws/issues/979#issuecomment-337934638
